SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -201,52 +201,89 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 
 	if (!bPersistSpatialConnection)
 	{
-		// Destroy the old connection
+		//Do we like these kinds of comments? The code seemed explanatory enough. (WILL BE REMOVED)
 		GameInstance->DestroySpatialWorkerConnection();
-
-		// Create a new SpatialWorkerConnection in the SpatialGameInstance.
 		GameInstance->CreateNewSpatialWorkerConnection();
 	}
 
 	Connection = GameInstance->GetSpatialWorkerConnection();
 
-	if (URL.HasOption(TEXT("locator")))
+	if (URL.Host == TEXT("locator.improbable.io"))
+	{
+		// use existing locator connection parameters
+		Connection->SetConnectionType(SpatialConnectionType::Locator);
+
+		// ensure that locator data was parsed from command line or that previous locator data exists
+		check(!Connection->LocatorConfig.LocatorHost.IsEmpty());
+
+	}
+	else if (URL.HasOption(TEXT("locator")))
 	{
 		// Obtain PIT and LT.
+		Connection->SetConnectionType(SpatialConnectionType::Locator);
 		Connection->LocatorConfig.PlayerIdentityToken = URL.GetOption(TEXT("playeridentity="), TEXT(""));
 		Connection->LocatorConfig.LoginToken = URL.GetOption(TEXT("login="), TEXT(""));
 		Connection->LocatorConfig.UseExternalIp = true;
 		Connection->LocatorConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
 	}
 	else // Using Receptionist
 	{
+		// Should SpatialConnectionType be ESpatialConnectionType? (WILL BE REMOVED)
```

Comment:
Also a good point, you can either make that change in the same PR or do a small follow up PR with just the rename.